src/Compilers/CSharp/Test/Syntax/Parsing/RawInterpolatedStringLiteralParsingTests.cs

```diff
@@ -0,0 +1,1465 @@
+                    Diagnostic(ErrorCode.ERR_StringMustStartWithQuoteCharacter, "$@@@").WithLocation(6, 17));
+    }
+
+    [Fact]
+    public void DollarThenAt6()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $@@@"";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics(
+            // (6,17): error CS9125: Cannot mix verbatim and raw strings
+            //         var v = $@@@";
+            Diagnostic(ErrorCode.ERR_IllegalAtSequence, @"$@@@""").WithLocation(6, 17),
+            // (6,23): error CS1002: ; expected
+            //         var v = $@@@";
+            Diagnostic(ErrorCode.ERR_SemicolonExpected, "").WithLocation(6, 23));
+    }
+
+    [Fact]
+    public void DollarThenAt7()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $@@@"" "";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics(
+            // (6,17): error CS9125: Cannot mix verbatim and raw strings
+            //         var v = $@@@" ";
+            Diagnostic(ErrorCode.ERR_IllegalAtSequence, @"$@@@""").WithLocation(6, 17));
+    }
+
+    [Fact]
+    public void DollarThenAt8()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $@@@"""""" """""";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics(
+            // (6,17): error CS9125: Cannot mix verbatim and raw strings
+            //         var v = $@@@""" """;
+            Diagnostic(ErrorCode.ERR_IllegalAtSequence, @"$@@@""""""").WithLocation(6, 17));
+    }
+
+    [Fact]
+    public void AtThenDollar1()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = @@$;
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics(
+                    // (6,17): error CS9125: String must start with quote character: "
+                    //         var v = @@$;
+                    Diagnostic(ErrorCode.ERR_StringMustStartWithQuoteCharacter, "@@$").WithLocation(6, 17));
+    }
+
+    [Fact]
+    public void AtThenDollar2()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = @@$"";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics(
+            // (6,17): error CS9125: Cannot mix verbatim and raw strings
+            //         var v = @@$";
+            Diagnostic(ErrorCode.ERR_IllegalAtSequence, @"@@$""").WithLocation(6, 17),
+            // (6,22): error CS1002: ; expected
+            //         var v = @@$";
+            Diagnostic(ErrorCode.ERR_SemicolonExpected, "").WithLocation(6, 22));
+    }
+
+    [Fact]
+    public void AtThenDollar3()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = @@$"" "";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics(
+            // (6,17): error CS9125: Cannot mix verbatim and raw strings
+            //         var v = @@$" ";
+            Diagnostic(ErrorCode.ERR_IllegalAtSequence, @"@@$""").WithLocation(6, 17));
```

Comment:
Consider doing a search&replace to fix the diagnostic message in tests (still shows "// (6,17): error CS9125: Cannot mix verbatim and raw strings" which is confusing)

Reply:
Gentle ping